erts/emulator/sys/common/erl_sys_common_misc.c

```diff
@@ -223,8 +223,9 @@ sys_double_to_chars_fast(double f, char *buffer, int buffer_size, int decimals,
                 } while (--decimals);
                 break;
             }
-            *p++ = (char)((frac_part % 10) + '0');
-            frac_part /= 10;
+            Uint64 n = frac_part / 10;
```

Comment:
We keep to pure C syntax, where variables have to be declared at the top of blocks.

Reply:
Corrected